nzshm_common/util/compression.py

```diff
@@ -0,0 +1,43 @@
+"""Useful utility & helpers."""
+import base64
+import io
+import zipfile
+from pathlib import Path, PurePath
+from typing import Union
+
+
+def compress_path(source_path: str, output_zip: Union[str, PurePath]) -> str:
+    """Zip contents of source path and return the full archive path.
+
+    Handles both single file and a folder.
+    """
+    with zipfile.ZipFile(output_zip, "w", compression=zipfile.ZIP_DEFLATED) as zip:
+        if Path(source_path).is_file():
+            zip.write(source_path, PurePath(source_path).name)
+        else:
+            for filename in Path(source_path).rglob("*"):
+                zip.write(filename, arcname=str(Path(filename).relative_to(source_path)))
+    return str(output_zip)
+
+
+def compress_string(config: str) -> str:
```

Comment:
it's not just for configs?

Reply:
@voj yep - I renamed the method, but missed the arg name.